### PR TITLE
Basic Models and DB

### PR DESCRIPTION
**Pull request by JPS09**
I double checked but feel free to look at the schema to be extra sure

### CODE REVIEW

**Comment by sbriclot** [comment]
BeerController :: Index
Beer.where(is_validated = true OR user_id = current_user.id)
==> select validated beer AND personal beers

BeerController :: Create
@beer.is_validated = false
==> only true or false for boolean, easier to use

BeerController :: Validate
Beer.where(is_validated = false)

File: app/controllers/beers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class BeersController < ApplicationController
+  skip_before_action :authenticate_user!, only: %i[index show]
+  before_action :set_beers, only: %i[show destroy edit update validate! decline!]
+
+  def index
+    @beers = Beer.where(:is_validated == true)
```
Comment (sbriclot) [inline app/controllers/beers_controller.rb]: Beer.where(is_validated == true OR user_id == current_user.id)

File: app/controllers/beers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  end
+
+  def new
+    @beer = Beer.new
+  end
+
+  def create
+    @beer.new(beers_params)
+    @beer.user = current_user
+    @beer.is_validated = nil # nil value == validation pending
```
Comment (sbriclot) [inline app/controllers/beers_controller.rb]: @beer.is_validated =false
(use only true or false for boolean)

File: app/controllers/beers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    redirect_to beer_path(@beer), notice: 'Beer successfully updated'
+  end
+
+  def destroy
+    @beer.destroy
+    redirect_to beers_path, notice: 'Beer successfully deleted'
+  end
+
+  def validation
+    @beers.where(:is_validated.nil?)
```
Comment (sbriclot) [inline app/controllers/beers_controller.rb]: Beer.where(is_validated == false)

File: app/controllers/beers_controller.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  def edit
+  end
+
+  def update
+    @beer.update(beers_params)
+    redirect_to beer_path(@beer), notice: 'Beer successfully updated'
+  end
+
+  def destroy
```
Comment (sbriclot) [inline app/controllers/beers_controller.rb:36]: Must be possible only for non-validated personal beers (maybe manage by Pundit)

File: app/controllers/breweries_controller.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    # need to geocode on save
+
+    if @bewery.save
+      redirect_to brewery_path(@brewery), notice: 'Brewery sucessfully created'
+    else
+      render :new
+    end
+  end
+
+  def destroy
```
Comment (sbriclot) [inline app/controllers/breweries_controller.rb:29]: Must be block if at least 1 beer of the brewery is assigned to 1 list.

File: app/controllers/beers_controller.rb (hunk 1 above)
Comment (JPS09) [reply app/controllers/beers_controller.rb]: Fixed

File: app/models/list.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class List < ApplicationRecord
+  belongs_to :user
+  has_many :contents
```
Comment (sbriclot) [inline app/models/list.rb]: add dependent :destroy to delete the list content at the same time

File: app/models/color.rb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+class Color < ApplicationRecord
+	has_many :beers
```
Comment (sbriclot) [inline app/models/color.rb]: tabulation

File: app/models/country.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class Country < ApplicationRecord
+	has_many :breweries
```
Comment (sbriclot) [inline app/models/country.rb]: tabulation

File: app/models/style.rb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+class Style < ApplicationRecord
+	has_many :beers
```
Comment (sbriclot) [inline app/models/style.rb]: tabulation

**Pull request merged**